src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryObjectInfo.cs

```diff
@@ -618,7 +618,8 @@ internal Type[] GetMemberTypes(string[] inMemberNames, Type objectType)
             bool isFound = false;
             for (int i = 0; i < _cache._memberInfos.Length; i++)
             {
-                if (!memberMissing && inMemberNames[i].Equals(_cache._memberInfos[i].Name))
+                if (!memberMissing && 
+                    inMemberNames[i].Equals(FormatterServices.GetFullMemberInfoName(_cache._memberInfos[i], objectType)))
```

Comment:
So basically we are now working with the FieldInfo type instead of the custom SerializationInfoField type in netfx. The SerializationInfoField took care of putting the base type name and the member name together. 
Therefore I added the code at the places where we are working with the FieldInfo types (creating a cache object and doing a verification check)